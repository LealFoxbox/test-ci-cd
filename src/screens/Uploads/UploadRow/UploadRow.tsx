import React from 'react';
import { View } from 'react-native';
import { Card, Paragraph, Title, useTheme } from 'react-native-paper';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import { isString } from 'lodash/fp';

import { styled } from 'src/paperTheme';
import { getAccessibilityAndAutomationProps } from 'src/utils/accessibility';

const Container = styled.View`
  background-color: white;
  padding: 10px;
`;

const ClickableContainer = styled.TouchableOpacity<{ disabled?: boolean }>`
  background-color: white;
  padding: 10px;
  opacity: ${(p) => (p.disabled ? 0.5 : 1)};
`;

interface RowProps {
  head: string;
  title: string;
  content: React.ReactNode;
  error?: string | null;
  accessibilityLabel?: string;
  icon: string;
  IconComponent?: typeof MaterialIcons;
  onPress?: () => void;
  spinner?: boolean;
  disabled?: boolean;
}

const Row: React.FC<RowProps> = ({
  head,
  title,
  content,
  error,
  accessibilityLabel,
  icon,
  IconComponent = MaterialIcons,
  disabled,
  onPress,
}) => {
  const theme = useTheme();
  const row = (
    <View style={{ flexDirection: 'row', alignItems: 'center', width: '100%', paddingLeft: 15 }}>
      {!error && <IconComponent name={icon} size={24} color={theme.colors.placeholder} />}
      {error && <MaterialIcons name="error" size={24} color={theme.colors.error} />}
      <Card.Content style={{ flex: 1 }}>
        <Paragraph>{head.toUpperCase()}</Paragraph>
        <Title>{title}</Title>
        {isString(content) ? <Paragraph>{content}</Paragraph> : content}
      </Card.Content>
    </View>
  );

  if (onPress) {
    return (
      <ClickableContainer
        onPress={onPress}
        accessibilityRole="button"
        disabled={disabled}
        {...getAccessibilityAndAutomationProps(accessibilityLabel || title)}
      >
        {row}
      </ClickableContainer>
    );
  }

  return <Container>{row}</Container>;
};

export default Row;
