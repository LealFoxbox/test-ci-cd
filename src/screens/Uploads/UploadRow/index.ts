export * from './UploadRow';
export { default } from './UploadRow';
