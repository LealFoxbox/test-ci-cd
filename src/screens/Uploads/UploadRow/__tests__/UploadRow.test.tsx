import React from 'react';
import { View } from 'react-native';
import { render } from '@testing-library/react-native';

import UploadRow from '../UploadRow';

describe('UploadRow', () => {
  it('should render', () => {
    const result = render(<UploadRow head="head" title="title" content={<View />} icon="user" />);

    expect(result).toMatchSnapshot();
  });
});
