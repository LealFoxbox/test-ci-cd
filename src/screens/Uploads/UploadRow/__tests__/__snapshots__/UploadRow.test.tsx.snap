// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadRow should render 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingLeft": 15,
        "width": "100%",
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontSize": 24,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      î½‚
    </Text>
    <View
      style={
        Array [
          Object {
            "paddingHorizontal": 16,
          },
          undefined,
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                undefined,
              ],
              Array [
                Object {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                undefined,
              ],
            ],
          ]
        }
      >
        HEAD
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "500",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 20,
                  "letterSpacing": 0.15,
                  "lineHeight": 30,
                  "marginVertical": 2,
                },
                undefined,
              ],
              Array [
                Object {
                  "fontSize": 20,
                  "letterSpacing": 0.15,
                  "lineHeight": 30,
                  "marginVertical": 2,
                },
                undefined,
              ],
            ],
          ]
        }
      >
        title
      </Text>
      <View />
    </View>
  </View>
</View>
`;
